internal/protocol/client_authentication.go

```diff
@@ -18,86 +18,88 @@ import (
 	. "github.com/hazelcast/hazelcast-go-client/internal/common"
 )
 
+type clientAuthenticationCodec struct {
+}
+
 type ClientAuthenticationResponseParameters struct {
 	Status                    uint8
 	Address                   *Address
 	Uuid                      *string
 	OwnerUuid                 *string
 	SerializationVersion      uint8
 	ServerHazelcastVersion    *string
-	ClientUnregisteredMembers *[]Member
+	ClientUnregisteredMembers []*Member
 }
 
-func ClientAuthenticationCalculateSize(username *string, password *string, uuid *string, ownerUuid *string, isOwnerConnection bool, clientType *string, serializationVersion uint8) int {
-	// Calculates the request payload size
-	dataSize := 0
-	dataSize += StringCalculateSize(username)
-	dataSize += StringCalculateSize(password)
+func (self *clientAuthenticationCodec) CalculateSize(args ...interface{}) (dataSize int) {
+	dataSize += StringCalculateSize(args[0].(*string))
+	dataSize += StringCalculateSize(args[1].(*string))
 	dataSize += BOOL_SIZE_IN_BYTES
-	if uuid != nil {
-		dataSize += StringCalculateSize(uuid)
+	if args[2].(*string) != nil {
+		dataSize += StringCalculateSize(args[2].(*string))
 	}
 	dataSize += BOOL_SIZE_IN_BYTES
-	if ownerUuid != nil {
-		dataSize += StringCalculateSize(ownerUuid)
+	if args[3].(*string) != nil {
+		dataSize += StringCalculateSize(args[3].(*string))
 	}
 	dataSize += BOOL_SIZE_IN_BYTES
-	dataSize += StringCalculateSize(clientType)
+	dataSize += StringCalculateSize(args[5].(*string))
 	dataSize += UINT8_SIZE_IN_BYTES
-	return dataSize
+	dataSize += StringCalculateSize(args[7].(*string))
+	return
 }
-
-func ClientAuthenticationEncodeRequest(username *string, password *string, uuid *string, ownerUuid *string, isOwnerConnection bool, clientType *string, serializationVersion uint8) *ClientMessage {
+func (self *clientAuthenticationCodec) EncodeRequest(args ...interface{}) (request *ClientMessage) {
```

Comment:
@sancar what do you think about this ☝️I think old codecs were better. This new approach does not bring much value.

Reply:
As far as I understand, this is to find a way to abstract these out. This way proxies will be simpler. 
Not sure which one I prefer.

Reply:
I think it is easier to write proxies with old codecs because type checking and IDE helps that you pass correct arguments to codecs. These new codecs eliminates only two lines from each proxy method (DecodeResponse, toObject). I think it is not worth it.

Reply:
I think writing codecs wont be hard either way, for sure it will take slightly more time. After finishing the proxies the code will look a little bit better since it will be more compacted. But honestly I am not sure which one is better for readibilty, with the old one we see everything in the method, with this one it is abstracted and cannot be understood that well.